File: app/src/main/java/com/jugger/tetris/Menu.java
package com.jugger.tetris;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.TextView;

import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;

public class Menu extends AppCompatActivity {

    private static int highscore = 0;
    public static final String FILE_NAME= "score.txt";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_menu);
        try {
            this.getSupportActionBar().hide();
        } catch (NullPointerException e) { }
        rotateFont();
        load_score();
    }

    public void startGame(View view) {
        Intent i = new Intent(this,MainActivity.class);
        startActivity(i);
    }

    public void  openSettings(View view) {
        Intent i = new Intent(this,SettingsActivity.class);
        startActivity(i);
    }

    public void rotateFont() {
        TextView headline =findViewById(R.id.m_tV_headline);
        headline.setRotation(0);
        Thread t = getThread_that_rotates_View(headline);

        t.start();
    }

    private  void load_score() {
        try  {
            FileInputStream fos = openFileInput(FILE_NAME);
            StringBuffer s_b = new StringBuffer();
            int a;
            while((a = fos.read()) > -1) {
                s_b.append((char)a);
            }
            String integer = s_b.toString();
            highscore = Integer.parseInt(integer);
            TextView score = findViewById(R.id.m_tV_highScore);
            score.setText("Highscore "+integer);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    private Thread getThread_that_rotates_View(final View view) {
        return new Thread(new Runnable() {
            @Override
            public void run() {
                Runnable runnable = getRunnable_that_rotates_View(view);
                while(true) {
                    runOnUiThread(runnable);
                    try {
                        Thread.sleep(10);
                    } catch(InterruptedException e) {

                    }
                }
            }
        });
    }

    private Runnable getRunnable_that_rotates_View(final View view) {
        return  new Runnable() {

            private int x = 1;

            public void change_direction() {
                x = x*-1;
            }

            @Override
            public void run() {
                view.setRotation(view.getRotation() +x);
                if(view.getRotation() == -40 || view.getRotation() == 40 ) {
                    change_direction();
                }
            }
        };
    }

    public static int getHighscore() {
        return highscore;
    }


}

File: app/src/main/java/com/jugger/tetris/Menu.kt
package com.jugger.tetris

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_menu.*


class Menu : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_menu)
        this.supportActionBar?.hide()
        rotateFont()

        var h =HighScore.getInstance(applicationContext.externalCacheDir)
        h.onHighscoreChanged = { score: Int ->
            runOnUiThread {
                m_tV_highScore.text = "Highscore $score"
            }
        }
        h.onHighscoreChanged(h.highscore)

    }

    fun startGame(view: View?) {
        val i = Intent(this, GameActivity::class.java)
        startActivity(i)
    }

    fun openSettings(view: View?) {}
    fun rotateFont() {
        val headline = m_tV_headline
        headline.rotation = 0f
        rotateView(headline).start()
    }

    private fun rotateView(view: View): Thread {
        return Thread(Runnable {
            val runnable = rotate(view)
            while (true) {
                runOnUiThread(runnable)
                try {
                    Thread.sleep(10)
                } catch (e: InterruptedException) {
                }
            }
        })
    }

    private fun rotate(view: View): Runnable {
        return object : Runnable {
            private var x = 1
            fun change_direction() {
                x = x * -1
            }
            override fun run() {
                view.rotation = view.rotation + x
                if (view.rotation == -40f || view.rotation == 40f) {
                    x = x * -1
                }
            }
        }
    }
}

